pkg/logging/options.go

```diff
@@ -116,3 +147,82 @@ var LogDecision = map[string]Decision{
 	"LogFinishCall":         LogFinishCall,
 	"LogStartAndFinishCall": LogStartAndFinishCall,
 }
+
+// MapAllowedLevels allows to map a given level to a list of allowed level.
+// Convention taken from go-kit/level v0.10.0 https://godoc.org/github.com/go-kit/kit/log/level#AllowAll.
+var MapAllowedLevels = map[string][]string{
+	"DEBUG": {"INFO", "DEBUG", "WARN", "ERROR"},
+	"ERROR": {"ERROR"},
+	"INFO":  {"INFO", "WARN", "ERROR"},
+	"WARN":  {"WARN", "ERROR"},
+}
+
+// TODO: @yashrsharma44 - To be deprecated in the next release.
+func DecideHTTPFlag(flagDecision string, configYAML []byte) ([]Option, error) {
+	// Check if the user enables request logging through flags and YAML
+	if len(configYAML) != 0 && len(flagDecision) != 0 {
+		return []Option{}, fmt.Errorf("Both log.request.decision and request.logging has been enabled. Please use one of the flags!")
+	}
+
+	// If old flag is enabled.
+	if len(flagDecision) > 0 {
+		logOpts := []Option{WithDecider(func(_ string, _ error) Decision {
+			return LogDecision[flagDecision]
+		})}
+		return logOpts, nil
+	}
+
+	logOpts, err := NewHTTPLoggingOption(configYAML)
+	if err != nil {
+		return []Option{}, err
+	}
+
+	return logOpts, nil
+}
+
+// TODO: @yashrsharma44 - To be deprecated in the next release.
+func DecideGRPCFlag(flagDecision string, configYAML []byte) ([]tags.Option, []grpc_logging.Option, error) {
+	// Check if the user enables request logging through flags and YAML
+	if len(configYAML) != 0 && len(flagDecision) != 0 {
+		return []tags.Option{}, []grpc_logging.Option{}, fmt.Errorf("Both log.request.decision and request.logging has been enabled. Please use one of the flags!")
+	}
+
+	if len(flagDecision) > 0 {
```

Comment:
It looks very indentend so maybe we can do opposite e.g `if len(flagDecision) == 0 {`

The many indents code is harder to read

Reply:
Sure